evalml/pipelines/components/transformers/column_selectors.py

```diff
@@ -122,3 +122,30 @@ def transform(self, X, y=None):
             pd.DataFrame: Transformed X.
         """
         return super().transform(X, y)
+
+
+class SelectByType(ColumnSelector):
+    """
+    Selects columns by specified Woodwork logical type or semantic tag in input data.
+
+    Arguments:
+        columns (list(string), list(ww.LogicalType)): List of datatypes, used to determine which columns to select.
+        random_seed (int): Seed for the random number generator. Defaults to 0.
+    """
+
+    name = "Select Columns By Type Transformer"
+    hyperparameter_ranges = {}
+    """{}"""
+    needs_fitting = False
+
+    def _check_input_for_columns(self, X):
+        """
+        This check is not required, since ww.select does not throw an error if the requested types are not in the dataset.
+        """
+        return
+
+    def _modify_columns(self, cols, X, y=None):
+        selected_columns = X.ww.select(cols)
+        if len(cols) > 0 and selected_columns.empty:
+            raise ValueError(f"Column(s) of type {cols} not found in input data")
```

Comment:
This is the same type of error we raise in `ColumnSelector._check_input_for_columns`. Do we need to define it in two places? I think the intent with the current layout of `ColumnSelector` methods was for `_check_input_for_columns` to do this sort of validation, and for `_modify_columns` to only include logic to make modifications (i.e. dropping columns for the `DropColumns` transformer)

I think we should define this error only in `_check_input_for_columns`.

If you want the error to include the column types for `SelectByType`, but omit the column types for `ColumnSelector` and other subclasses, I would suggest something like:
* Define a class attribute like `_VALIDATION_ERROR_FORMAT_STRING = "Columns {columns} not found in input data."`
* Override the value of `_VALIDATION_ERROR_FORMAT_STRING` in the `SelectByType` class definition to `"Columns {columns} of type {column_types} not found in input data"`
* Update `ColumnSelector._check_input_for_columns` to do
```python
raise ValueError(
                self._VALIDATION_ERROR_FORMAT_STRING.format(**self.parameters)
            )
```
* Delete `SelectByType._modify_columns`

Alternatively, we could simplify this, just delete `SelectColumns._modify_columns` and not have the error message say anything about types.

Make sense?